browser/src/layer/vector/CanvasOverlay.ts

```diff
@@ -485,6 +489,53 @@ class CanvasOverlay extends CanvasSectionObject {
 
 	}
 
+	setBoxGradient(path: CPath) {
+		const splitPos = this.tsManager.getSplitPos();
+		let selectionBackgroundGradient = null;
+
+		// last row geometry data will be a good for setting deafult raw height
+		const spanlist = this.map._docLayer.sheetGeometry.getRowsGeometry()._visibleSizes._spanlist;
+		const rowData = spanlist[spanlist.length - 1];
+
+		// Create a linear gradient based on the extracted color stops
+		// get raw data from sheet geometry. use index = 1
+		const deafultRowSize = rowData.data.sizecore;
+		// gradient width shoulb be half a default row hight.
+		const gradientWidth: number = Math.ceil(deafultRowSize / 2);
+		const colorStopsArray = [
```

Comment:
move this directly to `createLinearGradientBasedOnColorStops`
no need to pass it, it is used only there

Reply:
and then name could be eg. `createSplitLineGradient`